include/BSMPT/minimum_tracer/minimum_tracer.h

```diff
@@ -0,0 +1,830 @@
+Create1DimGrid(const std::vector<double> &point,
+               const int k,
+               const double low_value,
+               const double high_value,
+               const int nsteps = 100);
+
+/**
+ * @brief Create1DimGrid creates a 1-dim grid of given size between two points
+ * @param min_false
+ * @param min_true
+ * @param npoints
+ * @return npoints long vector of steps on connecting line between min_false and
+ * min_true
+ */
+std::vector<std::vector<double>>
+Create1DimGrid(const std::vector<double> &min_false,
+               const std::vector<double> &min_true,
+               const int npoints = 100);
+
+/**
+ * Returns true if two values are the same given some relative precision
+ */
+bool almost_the_same(const double &a,
+                     const double &b,
+                     const double &rel_precision = 0.01,
+                     const double &num_zero      = 1e-10);
+
+/**
+ * Returns true if two vectors are the element-wise the same given some relative
+ * precision
+ */
+bool almost_the_same(const std::vector<double> &a,
+                     const std::vector<double> &b,
+                     const bool &allow_for_sign_flip = false,
+                     const double &rel_precision     = 0.01,
+                     const double &num_zero          = 1e-10);
+
+/**
+ * @brief NoSignFlip project Minimum to positive quadrant in vev space
+ * @param a Minimum
+ */
+void NoSignFlip(Minimum &a);
+
+/**
+ * @brief NoSignFlip project Minimum to positive quadrant in vev space
+ * @param v std::vector<double>
+ */
+void NoSignFlip(std::vector<double> &v);
+
+/**
+ * @brief Phase object
+ *
+ */
+struct Phase
+{
+  /**
+   * @brief phase ID
+   */
+  int id = 0;
+
+  /**
+   * @brief Lowest temperature of the phase
+   */
+  double T_low = 0;
+
+  /**
+   * @brief Highest temperature of the phase
+   */
+  double T_high = 0;
+
+  /**
+   * @brief Set of Minimum that compose the phase
+   */
+  std::vector<Minimum> MinimumPhaseVector;
+
+  /**
+   * @brief MinTracer object
+   */
+  std::shared_ptr<MinimumTracer> MinTracer;
+
+  /**
+   * @brief Function that adds min to MinimumPhaseVector
+   *
+   * @param min Minimum
+   */
+  void Add(Minimum min);
+
+  /**
+   * @brief Calculates the minimum of the phase at temperature T. This function
+   * assumes that T is inside the temperature range of the phase.
+   *
+   * @param T temperature
+   * @return Minimum at temperature T for the corresponding phase
+   */
+  Minimum Get(double T);
+
+  /**
+   * @brief empty constructor
+   */
+  Phase();
+
+  /**
+   * @brief Construct a new Phase object with enforced global minimum tracing
+   * @param phase_start Initial starting point for the phase
+   * @param initialT Initial temperature
+   * @param finalT Final temperature
+   * @param globMinEndT Temperature where phase is first detected to no longer
+   * be global minimum
+   * @param MinTracerIn MinTracer pointer
+   */
+  Phase(std::vector<double> phase_start,
```

Comment:
```suggestion
  Phase(const std::vector<double>& phase_start,
```

Reply:
`phase_start` cannot be `const`. It can get altered by `ReduceVEV` and/or `ConvertToNonFlatDirections`.

Reply:
But you pass it by value, so you edit a copy. The changes are not passed through to the outside

Reply:
This comment also applies to the other ```Phase``` constructors and to the ```MinimumTracer::TrackPhase``` constructors. Although we edit a copy, the result is saved into the ```Vacuum``` class. We need a variable to store the point because we do some operations to it so we might as well use the function argument and keep the code clean.